code/modules/reagents/chemistry/reagents/medicine_reagents.dm

```diff
@@ -875,16 +875,37 @@
 
 /datum/reagent/medicine/mutadone
 	name = "Mutadone"
-	description = "Removes jitteriness and restores genetic defects."
+	description = "Removes jitteriness and temporarly supresses genes from manifesting, it also prevents new mutation from appearing while in the system."
 	color = "#5096C8"
 	taste_description = "acid"
+	overdose_threshold = 35
+	var/list/mut = list()
+
+/datum/reagent/medicine/mutadone/on_mob_metabolize(mob/living/carbon/M)
+	..()
+	var/len = M.dna.mutations.len
+	for(var/i in 1 to len)
+		mut += M.dna.mutations[i].type
 
 /datum/reagent/medicine/mutadone/on_mob_life(mob/living/carbon/M)
+	M.adjustStaminaLoss(0.5 *REM, 0)
 	M.jitteriness = 0
 	if(M.has_dna())
 		M.dna.remove_all_mutations(mutadone = TRUE)
 	if(!QDELETED(M)) //We were a monkey, now a human
 		..()
+/datum/reagent/medicine/mutadone/on_mob_end_metabolize(mob/living/carbon/M)
+	..()
+	M.dna.remove_all_mutations(mutadone = TRUE) //just to be sure
+	var/len = mut.len
+	for(var/i in 1 to len)
+		M.dna.add_mutation(mut[i])
```

Comment:
Is there a mutation that would reset progress from deleting/readding?

Reply:
what progress?

Reply:
That's what i'm asking, is there any mutation that would reset some sort of progression mechanic by del/add?

Reply:
stuff like add timers?

Reply:
not necessarily. Is there any perf-like mechanic where the mutation does something / changes in power over time?

I can't think of any atm so prob a nonissue.